File: UNIDADE/desktop_java_unidade/src/main/java/br/com/estufa/desktop/model/Alerta.java
package br.com.estufa.desktop.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public class Alerta {
    private Long id;
    private String tipo;        // TEMPERATURA, UMIDADE, SOLO
    private String severidade;  // CRITICO, ATENCAO
    private float valor;
    private float limiteMin;
    private float limiteMax;
    private String mensagem;
    private String geradoEm;    // ISO string da API

    public Alerta() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getTipo() { return tipo; }
    public void setTipo(String tipo) { this.tipo = tipo; }

    public String getSeveridade() { return severidade; }
    public void setSeveridade(String severidade) { this.severidade = severidade; }

    public float getValor() { return valor; }
    public void setValor(float valor) { this.valor = valor; }

    public float getLimiteMin() { return limiteMin; }
    public void setLimiteMin(float limiteMin) { this.limiteMin = limiteMin; }

    public float getLimiteMax() { return limiteMax; }
    public void setLimiteMax(float limiteMax) { this.limiteMax = limiteMax; }

    public String getMensagem() { return mensagem; }
    public void setMensagem(String mensagem) { this.mensagem = mensagem; }

    public String getGeradoEm() { return geradoEm; }
    public void setGeradoEm(String geradoEm) { this.geradoEm = geradoEm; }

    public boolean isCritico() { return "CRITICO".equalsIgnoreCase(severidade); }
}

File: UNIDADE/flutter_application_finalProject/lib/iot/Alerta.dart
class Alerta {
  final int? id;
  final String tipo; // TEMPERATURA, UMIDADE, SOLO
  final String severidade; // CRITICO, ATENCAO
  final double valor;
  final double limiteMin;
  final double limiteMax;
  final String? mensagem;
  final String? geradoEm;

  Alerta({
    this.id,
    required this.tipo,
    required this.severidade,
    required this.valor,
    required this.limiteMin,
    required this.limiteMax,
    this.mensagem,
    this.geradoEm,
  });

  bool get isCritico => severidade.toUpperCase() == 'CRITICO';

  factory Alerta.fromJson(Map<String, dynamic> json) {
    return Alerta(
      id: json['id'] as int?,
      tipo: (json['tipo'] ?? 'DESCONHECIDO').toString(),
      severidade: (json['severidade'] ?? 'ATENCAO').toString(),
      valor: (json['valor'] ?? 0).toDouble(),
      limiteMin: (json['limiteMin'] ?? 0).toDouble(),
      limiteMax: (json['limiteMax'] ?? 0).toDouble(),
      mensagem: json['mensagem']?.toString(),
      geradoEm: json['geradoEm']?.toString(),
    );
  }
}
